spec/features/user_updates_review_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+feature 'user updates review' do
+  # As a user
+  # I want to be able to update my own reviews
+  # because my original opinion was bad and I should feel bad
+  #
+  # Acceptance Criteria:
+  # [ ] User should be able to update any review from their profile page as
+  #     described below.
+  # [ ] When viewing my own review on a book's show page, there should be an
+  #     edit button next to that review only.
+  # [ ] The button leads to the review's show page, with pre-populated data.
+  # [ ] Upon successful form submission, A user is redirected to the book's show
+  #     page, with the message "Review updated successfully!"
+  # [ ] Upon unsuccessful form submission, the edit page is re-rendered with the
+  #     appropriate errors.
+
+  let!(:users_review) { FactoryGirl.create(:review) }
+  let!(:other_review) { FactoryGirl.create(:review, book: users_review.book) }
+
+  before(:each) { login_as(users_review.user, scope: :user) }
+
+  def expect_fields_autopopulated
+    expect(find_field('Title').value).to eq users_review.title
+    expect(find_field('Review').value).to eq users_review.body
+  end
+
+  def fill_in_form
+    fill_in 'Title', with: 'I hated it so much'
+    fill_in 'Review',
+            with: 'So much that flames were coming out the sides of my face'
+    select '4', from: 'Rating'
+    click_button 'Update Review'
```

Comment:
First, great work pulling stuff out into helper methods. It's good to get in the habit of abstracting (and extracting) logic out if it will be reused.

Second, and this is more my opinion: your method is called `fill_in_form`, not `complete_and_submit_form`. Maybe you could leave the `click_button 'Update Review'` out of the method.

So further down you could write:

```
click_button 'Edit'
expect_fields_autopopulated
fill_in_form
click_button 'Update Review'
```
Which to me gives a little more context and just reads more like the workflow you are testing.